torch/_dynamo/__init__.py

```diff
@@ -78,6 +78,8 @@ def reset() -> None:
         _reset_guarded_backend_cache()
         reset_frame_count()
         torch._C._dynamo.compiled_autograd.clear_cache()
+        convert_frame.FRAME_COUNTER = 0
```

Comment:
Not direct related to this PR, but we have to consolidate this frame counter with another one: ```curr_frame``` and the corresponding increase/decrease handling, I can submit a refactor later.
https://github.com/pytorch/pytorch/blob/834c7a1d3ea07878ad87d127ee28606fc140b552/torch/_dynamo/utils.py#L184-L198